send-transaction-service/src/send_transaction_service.rs

```diff
@@ -703,114 +476,17 @@ impl SendTransactionService {
             let wire_transactions = transactions
                 .iter()
                 .filter(|(signature, _)| batched_transactions.contains(signature))
-                .map(|(_, transaction_info)| transaction_info.wire_transaction.as_ref())
-                .collect::<Vec<&[u8]>>();
-
-            let iter = wire_transactions.chunks(config.batch_size);
-            for chunk in iter {
-                let mut addresses = config
-                    .tpu_peers
-                    .as_ref()
-                    .map(|addrs| addrs.iter().collect::<Vec<_>>())
-                    .unwrap_or_default();
-                let mut leader_info_provider = leader_info_provider.lock().unwrap();
-                let leader_info = leader_info_provider.get_leader_info();
-                let leader_addresses = Self::get_tpu_addresses(
-                    tpu_address,
-                    leader_info,
-                    config,
-                    connection_cache.protocol(),
-                );
-                addresses.extend(leader_addresses);
-
-                for address in &addresses {
-                    Self::send_transactions(address, chunk, connection_cache, stats);
-                }
+                .map(|(_, transaction_info)| transaction_info.wire_transaction.clone());
+
+            let iter = wire_transactions.chunks(batch_size);
+            for chunk in &iter {
+                let chunk = chunk.collect();
+                client.send_transactions_in_batch(chunk, stats);
```

Comment:
Before we had a copy-pasted here, the same as in `send_transactions_in_batch` except for not using slots. I removed version of the code with slots and use the same method everywhere now.